README.md

```diff
@@ -168,6 +168,15 @@ config :ueberauth, Ueberauth,
   ]
 ```
 
+## Customizing JSON Serializer
+
+Your json encoder/decoder can be configured.  Defaults to Poison in other ueberauth libraries such as [ueberauth_google](https://github.com/ueberauth/ueberauth_google) and [ueberauth_twitter](https://github.com/ueberauth/ueberauth_twitter).  If you only have [Jason](https://github.com/michalmuskala/jason) installed, then it is a good idea to take advantage of this configuration property.
+
+```elixir
+config :ueberauth, Ueberauth,
```

Comment:
@snewcomer ^

Reply:
@yordis One question I had was would the config be in the same block as the `provider` config?  Should we move all the documentation to be `config :ueberauth,` or will the other configs stay the same `config :ueberauth, Ueberauth,`?

Reply:
🤔 I follow some rules based on my personal experience.

I put the configuration to the closest context. In this case `config :ueberauth, Config` and then `config :ueberauth` **if and only if** there is no way to have more than one `Ueberauth` module, in case that happen I put it on `config :ueberauth, Ueberauth` instead of `config :ueberauth`.

If there is no context for the config I put it on the package name.

> Should we move all the documentation to be config :ueberauth, or will the other configs stay the same config :ueberauth, Ueberauth,?

This will introduce breaking changes or at least more code to maintain the backward compatibility. I wouldn't do it.


What I would do in this case is to move `json_library/0` to `Ueberauth` instead of introducing another module for this.

And yes keep the config in `config :ueberauth, Ueberauth` so we keep it consitent with other configs.

Reply:
No valid